packages/lib-classifier/src/components/Classifier/components/QuickTalk/QuickTalkContainer.js

```diff
@@ -0,0 +1,295 @@
+      const user = await authClient.checkCurrent()
+      if (!authorization || !user) throw('User not logged in')
+
+      // First, get default board
+      const boards = await talkClient.type('boards').get({ section, subject_default: true })
+      const defaultBoard = boards && boards[0]
+      if (!defaultBoard) throw('A board for subject comments has not been setup for this project yet.')
+      
+      // Next, attempt to find if the Subject already has a discussion attached to it.
+      const discussions = await talkClient.type('discussions').get({
+        board_id: defaultBoard.id,
+        title: discussionTitle,
+        subject_default: true,
+      })
+      const existingDiscussion = discussions && discussions[0]
+      
+      if (existingDiscussion) { // Add to the existing discussion
+
+        const comment = {
+          user_id: user.id,
+          body: text,
+          discussion_id: +existingDiscussion.id,
+        }
+
+        await talkClient.type('comments').create(comment).save()
+        this.setState({
+          postCommentStatus: asyncStates.success,
+          postCommentStatusMessage: '',
+        })
+        this.fetchComments()
+
+      } else {  // Create a new discussion
+
+        const comments = [{
+          user_id: user.id,
+          body: text,
+          focus_id: +subject.id,
+          focus_type: 'Subject',
+        }]
+
+        const discussion = {
+          title: discussionTitle,
+          user_id: user.id,
+          subject_default: true,
+          board_id: defaultBoard.id,
+          comments: comments,
+        }
+
+        await talkClient.type('discussions').create(discussion).save()
+        this.setState({
+          postCommentStatus: asyncStates.success,
+          postCommentStatusMessage: '',
+        })
+        this.fetchComments()
+      }
+
+    } catch (err) {
+      console.error(err)
+      this.setState({
+        postCommentStatus: asyncStates.error,
+        postCommentStatusMessage: err?.message || err,
+      })
+    }
+  }
+    
+  render () {
+    const {
+      subject,
+      enabled,
+    } = this.props
+    
+    const {
+      comments,
+      authors,
+      authorRoles,
+      postCommentStatus,
+      postCommentStatusMessage,
+      userId,
+    } = this.state
+
+    if (!subject || !enabled) {
+      return null
+    }
+    
+    return (
+      <QuickTalk
+        subject={subject}
+        comments={comments}
+        authors={authors}
+        authorRoles={authorRoles}
+        postCommentStatus={postCommentStatus}
+        postCommentStatusMessage={postCommentStatusMessage}
+        postComment={this.postComment.bind(this)}
+        userId={userId}
+      />
+    )
+  }
+}
+
+QuickTalkContainer.propTypes = {
+  authClient: PropTypes.object, // Quick Fix
+  enabled: PropTypes.bool,
+  subject: PropTypes.object,
+}
+
+QuickTalkContainer.defaultProps = {
+  authClient: undefined,
+  enabled: false,
+  subject: undefined,
+}
+
+@inject(storeMapper)
```

Comment:
Check out `helpers/withStores`. It’s a helper I wrote to replace `inject` in new components.